File: 167-Two_Sum_II/main.cpp
//
// Created by miller on 3/20/17.
//
#include <iostream>
#include <vector>

using namespace std;

vector<int> twoSum(vector<int>& numbers, int target) {
  int i = 0, j = numbers.size() - 1;
  for(; i != j; ){
    if(numbers[i] + numbers[j] > target)
      --j;
    else if(numbers[i] + numbers[j] < target)
      ++i;
    else
      break;
  }
  if( i != j)
    return {i,j};
  else
    return {-1, -1};
    //不存在
}
template <class T>
void print(vector<T> vec){
  for(auto& iter:vec){
    cout << iter << ' ';
  }
  //刷新缓冲区
  cout << endl;
}
int main(int argc, char** argv){
  //Test Case
  vector<int> numbers{2, 7, 11, 15};
  auto res = twoSum(numbers, 22);
  print<int>(res);
  return 0;
}

File: 167-Two_Sum_II/main.py
#coding=utf-8

def twoSum(numbers, target):
    i,j = 0,len(numbers) - 1
    while i!=j:
        if numbers[i] + numbers[j] > target:
            j-=1
        elif numbers[i] + numbers[j] < target:
            i+=1
        else:
            break
    if i!=j:
        return [i+1,j+1]
    else:
        raise ValueError("Not as expected.")

if __name__ == '__main__':
    print twoSum([2,7,11,15], 9)
